From 216a742cda811b39c6ba7df3195db5a53a834e71 Mon Sep 17 00:00:00 2001
From: Marc-Antoine Perennou <keruspe@exherbo.org>
Date: Fri, 6 Dec 2019 14:36:21 +0100
Subject: [PATCH] vfs0090

Signed-off-by: Marc-Antoine Perennou <keruspe@exherbo.org>
---
 packages/base/libfprint/libfprint-1.0.exheres-0 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/packages/base/libfprint/libfprint-1.0.exheres-0 b/packages/base/libfprint/libfprint-1.0.exheres-0
index 30ea96c..4c0eeb5 100644
--- a/packages/base/libfprint/libfprint-1.0.exheres-0
+++ b/packages/base/libfprint/libfprint-1.0.exheres-0
@@ -32,6 +32,7 @@ MYOPTIONS="
             uru4000 [[ description = [ Digital Personal U.are.U 4000/4000B ] ]]
             vcom5s [[ description = [ Veridicom 5thSense ] ]]
             vfs0050 [[ description = [ Validity VFS0050+ ] ]]
+            vfs0090 [[ description = [ Validity VFS0090+ ] ]]
             vfs101 [[ description = [ Validity VFS101 ] ]]
             vfs301 [[ description = [ Validity VFS301/VFS300 ] ]]
             vfs5011 [[ description = [ Validity VFS5011 ] ]]
-- 
2.24.0

