From 0135ea15c28365d93b9230ef3966b93142d9cce5 Mon Sep 17 00:00:00 2001
From: Marc-Antoine Perennou <Marc-Antoine@Perennou.com>
Date: Fri, 21 Oct 2011 12:02:22 +0200
Subject: [PATCH] Tmp fix

---
 src/main.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/src/main.c b/src/main.c
index b7c0fd5..f2eca92 100644
--- a/src/main.c
+++ b/src/main.c
@@ -563,8 +563,6 @@ main (int argc, char *argv[])
 	umask (022);
 
 	g_type_init ();
-	if (!g_thread_supported ())
-		g_thread_init (NULL);
 	dbus_g_thread_init ();
 
 #ifndef HAVE_DBUS_GLIB_DISABLE_LEGACY_PROP_ACCESS
-- 
1.7.7.419.g87009.dirty

